src/libcore/array.rs

```diff
@@ -406,6 +408,73 @@ mod impls_using_const_generics {
         }
     }
 
+    #[unstable(feature = "array_value_iter", issue = "0")]
+    impl<T, const N: usize> IntoIterator for [T; N]
+    where
+        [T; N]: LengthAtMost32,
+    {
+        type Item = T;
+        type IntoIter = IntoIter<T, {N}>;
+
+        fn into_iter(self) -> Self::IntoIter {
+            IntoIter {
+                data: ManuallyDrop::new(self),
+                pos: 0,
+            }
+        }
+    }
+
+    /// A by-value array iterator.
```

Comment:
```suggestion
    /// A by-value [array] iterator.
```
(and link to https://doc.rust-lang.org/nightly/std/primitive.array.html)